### PR TITLE
Make our Default Methods respect the Default options a little bit better

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#42470](https://central.tri.be/issues/42470)

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -3042,10 +3042,20 @@ protected function dateHelper( $date ) {
...
+			// If the post doesn't exist just bail the get_post_meta
+			if ( is_string( $status ) && 'auto-draft' !== $status ) {
+				$value = get_post_meta( $id, $meta, $single );
+			} else {
+				$value = false;
+			}
+
 			if ( $value === false ) {
-				$method = str_replace( '_Event', '', $meta );
+				$method = str_replace( array( '_Event', '_Organizer', '_Venue' ), '', $meta );
```
Comment (borkweb) [inline src/Tribe/Main.php]: There's a few duplicated `$metaTags`/`$venueTags`/`$organizerTags` that - when the prefix is stripped off of all three types - will result in identical filters below. Example:

* `_EventURL`
* `_VenueURL`
* `_OrganizerURL`

When stripped of the prefix, they all become just `URL`. Then the filter on line 3059 will be `filter_eventsDefaultURL` for all three, giving none of them distinction.

Is that intended?

Comment (bordoni) [inline src/Tribe/Main.php]: I don't think so, didn't think of that angle. might need to refactor the defaults class

File: src/Tribe/Main.php [hunk 2]
```diff
@@ -3042,11 +3042,43 @@ protected function dateHelper( $date ) {
...
+				/**
+				 * Used to Filter the default value for a Specific meta
+				 *
+				 * @since 4.0.7
+				 * @var $default
+				 * @var $id
+				 * @var $meta
+				 * @var $single
+				 */
+				$value = apply_filters( 'tribe_get_meta_default_value_' . $filter, $default, $id, $meta, $single );
```
Comment (borkweb) [inline src/Tribe/Main.php]: Should `$default` change to `$value` here?

**Comment by bordoni** [comment]
Ready for CR again :crystal_ball:

**Comment by borkweb** [comment]
:octopus:

**Pull request merged**